WordPress/Classes/Models/Blog.m

```diff
@@ -340,9 +341,79 @@ - (BOOL)supportsFeaturedImages
     return NO;
 }
 
+- (BOOL)supports:(BlogFeature)feature
+{
+    switch (feature) {
+        case BlogFeatureRemovable:
+            return ![self accountIsDefaultAccount];
+        case BlogFeatureVisibility:
+            /*
+             See -[BlogListViewController fetchRequestPredicateForHideableBlogs]
+             If the logic for this changes that needs to be updated as well
+             */
+            return [self accountIsDefaultAccount];
+        case BlogFeatureWPComRESTAPI:
+        case BlogFeatureLikes:
+            return [self restApi] != nil;
+        case BlogFeatureStats:
+            return [self restApiForStats] != nil;
+        case BlogFeatureCommentLikes:
+        case BlogFeatureReblog:
+        case BlogFeatureMentions:
+        case BlogFeatureOAuth2Login:
+            return [self isHostedAtWPcom];
+        case BlogFeaturePushNotifications:
+            return [self supportsPushNotifications];
+        default:
```

Comment:
Hmm, I actually added the default case because for some reason I thought the compiler would complain about not having it.